dspace-api/src/main/java/org/dspace/app/suggestion/runnable/PublicationLoaderRunnable.java

```diff
@@ -121,6 +121,8 @@ public void internalRun() throws Exception {
                 Item researcher = researchers.next();
                 publicationLoader.importRecords(context, researcher);
                 setLastImportMetadataValue(researcher);
+                context.commit();
+                context.uncacheEntity(researcher);
```

Comment:
I think the problem here may that you've closed the database transaction which was connected to `researchers` and `researcher`.  You might need to do a `context.reloadEntity(researcher);` above right after you call `Item researcher = researchers.next()` in order to reconnect it to the current context.

For example, see how a similar process is managed here https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/identifier/doi/DOIOrganiser.java#L228-L239